LetsBTF.sol

```diff
@@ -0,0 +1,759 @@
+        }
+
+        int256 value = int256(msg.value);
+        unchecked {
+            if (credit.claimedCredit - value == 0) {
+                // Avoid setting claimedCredit back to 0 to save gas (~15k)
+                // the next time credit is claimed
+                credit.claimedCredit = -1;
+            } else {
+                credit.claimedCredit -= value;
+            }
+        }
+    }
+
+    /**
+     * @notice Determines if a token is locked from being transferred.
+     * This is true if any credit is claimed for the token.
+     * @param _tokenId The ID of the token to check the lock status of
+     */
+    function isTokenLocked(uint256 _tokenId) public view returns (bool) {
+        return (tokenCredits[_tokenId].claimedCredit > 0);
+    }
+
+    /* =========== INCREASING CREDIT FUNCTIONS =========== */
+
+    /**
+     * @notice Increase the credit of a specific token by 
+     * sending ether to this function
+     * @param _id The ID of the token you wish to increase the credit of
+     * @dev Anyone can call this function sending ether and 
+     *     a token ID to increase the credit - you do not have to own the token.
+     */
+    function increaseCredit(uint256 _id)
+        external
+        payable
+        amountGreaterThanZero(msg.value)
+        verifyTokenExists(_id)
+    {
+        Credit storage credit = tokenCredits[_id];
+
+        // Initial credit has not yet been set via Chainlink VRF `fulfillRandomWords`
+        if (credit.totalCredit == 0) {
+            revert InitialCreditNotSet();
+        }
+
+        credit.totalCredit += msg.value;
+
+        emit CreditIncreased(block.timestamp, _id, msg.value);
+    }
+
+    /**
+     * @dev Use this to increase multiple token credits at once.
+     * The amount of ether sent will be equally divided amongst
+     * the token ID's passed in.
+     * @param _ids The IDs of the tokens to increase the credit for
+     */
+    function increaseCreditBatched(uint256[] calldata _ids)
+        external
+        payable
+        onlyRole(CREDIT_INCREASOR_ROLE)
+        amountGreaterThanZero(msg.value)
+    {
+        if (_ids.length == 0) {
+            revert NoTokenIdsGiven();
+        }
+
+        uint256 amountToIncrease = msg.value / _ids.length;
+
+        for (uint256 i = 0; i < _ids.length; ++i) {
+            uint256 tokenId = _ids[i];
+
+            if (!_tokenExists(tokenId)) {
+                revert TokenDoesNotExist();
+            }
+
+            tokenCredits[tokenId].totalCredit += amountToIncrease;
+        }
+    }
+
+    /**
+     * @dev Use this to increase every single cards' credit equally.
+     * The `additionalCreditPerCard` value is added to the cards
+     * total credit in all credit calculations.
+     */
+    function increaseAllCredit()
+        external
+        payable
+        onlyRole(CREDIT_INCREASOR_ROLE)
+        amountGreaterThanZero(msg.value)
+    {
+        additionalCreditPerCard += msg.value / MAX_SUPPLY;
```

Comment:
Add a require to ensure that `msg.value` is a multiple of `MAX_SUPPLY` to prevent rounding errors;